esp-hal/src/systimer.rs

```diff
@@ -45,31 +46,93 @@ use crate::{
 
 // TODO this only handles unit0 of the systimer
 
-pub struct SystemTimer<'d> {
-    _inner: PeripheralRef<'d, SYSTIMER>,
-    pub alarm0: Alarm<Target, 0>,
-    pub alarm1: Alarm<Target, 1>,
-    pub alarm2: Alarm<Target, 2>,
-}
+/// The SystemTimer
+pub struct SystemTimer;
 
-impl<'d> SystemTimer<'d> {
+impl SystemTimer {
+    /// Bitmask to be applied to the raw register value
     #[cfg(esp32s2)]
     pub const BIT_MASK: u64 = u64::MAX;
     #[cfg(not(esp32s2))]
     pub const BIT_MASK: u64 = 0xF_FFFF_FFFF_FFFF;
 
+    /// The ticks per second the underlying peripheral uses
     #[cfg(esp32s2)]
     pub const TICKS_PER_SECOND: u64 = 80_000_000; // TODO this can change when we have support for changing APB frequency
     #[cfg(not(esp32s2))]
     pub const TICKS_PER_SECOND: u64 = 16_000_000;
 
-    pub fn new(p: impl Peripheral<P = SYSTIMER> + 'd) -> Self {
+    /// Create a new instance in [crate::Blocking] mode.
+    ///
+    /// You can optionally pass interrupt handlers to be registered
+    pub fn create<'d>(
```

Comment:
I introduced the intermediate step of creating a `SystemTimers` instance to avoid specifying the driver-mode. Clippy suggested `new` should only create `Self` so I renamed it to `create`